progress-bar.js

```diff
@@ -0,0 +1,93 @@
+(function(){
+    if (!document.addEventListener ||
+        (  document.documentElement.style.transform === undefined
+        && document.documentElement.style.webkitTransform === undefined)){
+        return;
+    }
+
+    var TRANSFORM = document.documentElement.style.webkitTransform !== undefined ? 'webkitTransform' : 'transform';
+
+    var options = INSTALL_OPTIONS;
+
+    var containerEl = null;
+    var barEl = null;
+    var render = function(){
+        if (!document.body)
+            return;
+
+        if (!containerEl){
+            containerEl = document.createElement('div');
+            document.body.appendChild(containerEl);
+
+            barEl = document.createElement('div');
+            barEl.className = 'scroll-progress-bar-bar';
+            containerEl.appendChild(barEl);
+        }
+
+        containerEl.className = 'scroll-progress-bar-container';
+        containerEl.className += ' scroll-progress-bar-' + options.position;
+
+        barEl.style.height = options.thickness + "px";
+        barEl.style.backgroundColor = options.barColor;
+
+        if (options.containerHasColor)
+            containerEl.style.backgroundColor = options.containerColor;
+        else
+            containerEl.style.backgroundColor = '';
+
+        updateProgress();
+    };
+
+    var now = function(){
+        return (window.performance && performance.now()) || (+ new Date);
+    };
+
+    var lastCall = 0;
+    var updateProgress = function(){
+        if (!barEl)
+            return;
+
+        // Don't allow updates faster than 60 FPS
```

Comment:
Some older browsers have a tendency to trigger `scroll` events very frequently during scrolling, much more frequently than we need.  Running the JS during every event can slow down the browser.  This code prevents this update from occurring faster than is necessary for a smooth experience.